packages/protocol/contracts/governance/Governance.sol

```diff
@@ -859,93 +834,107 @@ contract Governance is
-      return ProposalStage.Execution;
-    // solhint-disable-next-line not-rely-on-time
-    } else if (now >= stageStartTime(dequeueTime, ProposalStage.Referendum)) {
-      return ProposalStage.Referendum;
-    } else {
-      return ProposalStage.Approval;
-    }
+  function isDequeuedProposal(
+    Proposals.Proposal storage proposal,
+    uint256 proposalId,
+    uint256 index
+  )
+    private
+    view
+    returns (bool)
+  {
+    return proposal.exists() && dequeued[index] == proposalId;
   }
 
   /**
-   * @notice Returns the starting time for a particular stage.
-   * @param dequeueTime The timestamp of when the proposal was dequeued.
-   * @param stage The stage to return the start time for.
-   * @return The stage start time.
+   * @notice Returns whether or not a dequeued proposal has expired.
+   * @param proposal The proposal struct.
+   * @return Whether or not the dequeued proposal has expired.
    */
-  function stageStartTime(uint256 dequeueTime, ProposalStage stage) public view returns (uint256) {
-    if (stage == ProposalStage.Approval) {
-      return dequeueTime;
-    } else if (stage == ProposalStage.Referendum) {
-      return dequeueTime.add(stageDurations.approval);
-    } else if (stage == ProposalStage.Execution) {
-      return dequeueTime.add(stageDurations.approval).add(stageDurations.referendum);
-    } else if (stage == ProposalStage.Expiration) {
-      return dequeueTime.add(stageDurations.approval).add(stageDurations.referendum).add(
-        stageDurations.execution
-      );
-    } else {
-      require(false);
-    }
+  function isDequeuedProposalExpired(
+    Proposals.Proposal storage proposal,
+    Proposals.Stage stage
+  )
+    private
+    view
+    returns (bool)
+  {
+    // The proposal is considered expired under the following conditions:
+    //   1. Past the approval stage and not approved.
+    //   2. Past the referendum stage and not passing.
+    //   3. Past the execution stage.
+    return (
+      (stage > Proposals.Stage.Execution) ||
+      (stage > Proposals.Stage.Referendum && !_isProposalPassing(proposal)) ||
+      (stage > Proposals.Stage.Approval && !proposal.isApproved())
+    );
   }
 
   /**
    * @notice Deletes a dequeued proposal.
+   * @param proposal The proposal struct.
    * @param proposalId The ID of the proposal to delete.
    * @param index The index of the proposal ID in `dequeued`.
    */
-  function deleteDequeuedProposal(uint256 proposalId, uint256 index) private {
+  function deleteDequeuedProposal(
+    Proposals.Proposal storage proposal,
+    uint256 proposalId,
+    uint256 index
+  )
+    private
+  {
+    if (proposal.isApproved() && proposal.totalWeight > 0) {
+      updateParticipationBaseline(proposal);
+    }
     dequeued[index] = 0;
     emptyIndices.push(index);
     delete proposals[proposalId];
   }
 
+  /**
+   * @notice Updates the participation baseline using the total number of votes on the proposal
```

Comment:
nit: consider `Updates the participation baseline based on the proportion of BondedDeposit weight that participated in the proposal's Referendum stage.`